Add Open Arabic LLM Leaderboard Benchmarks (Full and Light Version)

Malikeh97 opened the pull request:
Contributors: @shahrzads @Malikeh97

In this PR, we fully implemented and tested two following task sets and added them to `lm-evaluation-harness` repo.

- All the benchmarks under the Open Arabic LLM Leaderboard according to their lighteval implementation: [Link](https://huggingface.co/spaces/OALL/Open-Arabic-LLM-Leaderboard)

     -       The tasks are under `lm_eval/tasks/arabic_leaderboard_complete` folder

- The light version of the Open Arabic LLM Leaderboard benchmark with **10%** of the train sets and full validation sets for quick evaluation of the models. 

     -       The tasks are under `lm_eval/tasks/arabic_leaderboard_light` folder
   Note! The further details of groups, tasks, and subtasks are added as a README file to both folders.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2232) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2232) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: shahrzads<br/>:x: Malikeh97<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EleutherAI/lm-evaluation-harness?pullRequest=2232) it.</sub>

haileyschoelkopf commented on .gitignore:
Let's remove this change from .gitignore.

haileyschoelkopf commented on lm_eval/tasks/README.md:
```suggestion
| [arabic_leaderboard_complete](arabic_leaderboard_complete/README.md) | A full version of the tasks in the Open Arabic LLM Leaderboard focusing on the evaluation of models that reflect the characteristics of Arabic language understanding and comprehension, culture, and heritage. | Arabic |
```

haileyschoelkopf commented on lm_eval/tasks/README.md:
```suggestion
| [arabic_leaderboard_light](arabic_leaderboard_light/README.md) | A light version of the tasks in the Open Arabic LLM Leaderboard (i.e., 10% samples of the training set in the original benchmarks) focusing on the evaluation of models that reflect the characteristics of Arabic language understanding and comprehension, culture, and heritage. | Arabic |
```

haileyschoelkopf commented on lm_eval/tasks/README.md:
Could we also make more clear that some of these tasks are machine-translated?

haileyschoelkopf commented on lm_eval/tasks/arabic_leaderboard_complete/README.md:
How does this compare to the tasks in https://github.com/EleutherAI/lm-evaluation-harness/tree/main/lm_eval/tasks/arabicmmlu ?

haileyschoelkopf commented on lm_eval/tasks/arabic_leaderboard_complete/README.md:
Let's indicate in the names that these tasks are machine-translated--

`arabic_mt_hellaswag` etc?

haileyschoelkopf commented on lm_eval/tasks/arabic_leaderboard_complete/README.md:
Can these tasks include `alghafa` in their task names? these names seem a bit generic

haileyschoelkopf commented on lm_eval/tasks/arabic_leaderboard_complete/README.md:
```suggestion
      - `acva_Algeria`
```
These task names are not descriptive enough--how should users know they are part of acva? could we change the names?

haileyschoelkopf commented on lm_eval/tasks/arabic_leaderboard_complete/README.md:
Does this overlap with https://github.com/EleutherAI/lm-evaluation-harness/tree/main/lm_eval/tasks/alghafa ? should we deprecate those in favor of this variant?

haileyschoelkopf reviewed (changes requested):
Hi, thanks very much for the substantial contribution!

I've left a review--the main comment I have is that 1) the task names should be more clearly named, and 2) also, we should be very clear what tasks are machine-translated and what are not.

(perhaps using a prefix like `arabic_leaderboard` for these tasks' names is also desirable?)

shahrzads commented on .gitignore:
Done

shahrzads commented on lm_eval/tasks/README.md:
done

shahrzads commented on lm_eval/tasks/README.md:
Done

shahrzads commented on lm_eval/tasks/arabic_leaderboard_complete/README.md:
added indicator for machine translated benchmarks and their files - for both light and full versions

shahrzads commented on lm_eval/tasks/arabic_leaderboard_complete/README.md:
Added `alghafa` to the related tasks for both the light and full versions

shahrzads commented on lm_eval/tasks/arabic_leaderboard_complete/README.md:
added `acva` to the name of the related tasks for both the light and full versions

shahrzads commented on lm_eval/tasks/arabic_leaderboard_complete/README.md:
Our goal in the provided benchmarks is to replicate the same results as in the Open Arabic LLM Leaderboard. To this end, the benchmarks (full and light) are completely based on what is mentioned in Open Arabic LLM Leaderboard (https://huggingface.co/spaces/OALL/Open-Arabic-LLM-Leaderboard). In this leaderboard, the evaluations are based on lighteval (https://github.com/huggingface/lighteval). We used the same datasets, benchmarks, and procedure from lighteval to evaluate the models with complete alignment in lm evaluation harness as well.
The datasets and process_doc procedure in  https://github.com/EleutherAI/lm-evaluation-harness/tree/main/lm_eval/tasks/arabicmmlu does not match with what is mentioned in lighteval.

shahrzads commented on lm_eval/tasks/arabic_leaderboard_complete/README.md:
The dataset in https://github.com/EleutherAI/lm-evaluation-harness/tree/main/lm_eval/tasks/alghafa is more or less the same as the dataset we used, except it only contains data in test split (no data in validation, which is used for few shot evaluation). Also, we followed the process_doc to be exactly the same as what is available in lighteval (https://github.com/huggingface/lighteval). Our module contains all the benchmarks with the exact details and datasets as in the Open Arabic LLM Leaderboard and lighteval.

Malikeh97 commented:
> Hi, thanks very much for the substantial contribution!
> 
> I've left a review--the main comment I have is that 1) the task names should be more clearly named, and 2) also, we should be very clear what tasks are machine-translated and what are not.
> 
> (perhaps using a prefix like `arabic_leaderboard` for these tasks' names is also desirable?)

Hi @haileyschoelkopf ,
My colleague @shahrzads addressed all your comments and we are now retesting the finalized benchmarks on some sample models. 
Thanks a lot for your constructive feedback and thanks for open-sourcing such a fantastic repo! It was our pleasure to contribute!

Malikeh97 commented:
> Hi, thanks very much for the substantial contribution!
> 
> I've left a review--the main comment I have is that 1) the task names should be more clearly named, and 2) also, we should be very clear what tasks are machine-translated and what are not.
> 
> (perhaps using a prefix like `arabic_leaderboard` for these tasks' names is also desirable?)

Hi @haileyschoelkopf 
Thanks for you constructive comments. My colleague @shahrzads addressed all the comments and fully tested both light and full versions on sample models. It would be great if you do a final review of the PR and let us know if anything is still missing. 

Thanks for creating such a fantastic repo and it was our pleasure to contribute!

shahrzads commented:
> Hi, thanks very much for the substantial contribution!
> 
> I've left a review--the main comment I have is that 1) the task names should be more clearly named, and 2) also, we should be very clear what tasks are machine-translated and what are not.
> 
> (perhaps using a prefix like `arabic_leaderboard` for these tasks' names is also desirable?)

Hi @haileyschoelkopf,
Thank you for your thorough review and valuable feedback. We have carefully addressed all the comments and made the necessary adjustments to the code. We also tested both light and full versions on sample models. Please let us know if there are any further changes required. Otherwise, we believe the updates are ready for merge and look forward to your approval. Your prompt action would be greatly appreciated.

Thanks for the great repo @haileyschoelkopf, @lintangsutawika, and @baberabb. It was a pleasure to contribute to this amazing project!

haileyschoelkopf commented:
Thank you very much @shahrzads @Malikeh97 , I will review again today!

haileyschoelkopf commented on lm_eval/tasks/README.md:
```suggestion
| [arabic_leaderboard_complete](arabic_leaderboard_complete/README.md) | A full version of the tasks in the Open Arabic LLM Leaderboard, focusing on the evaluation of models that reflect the characteristics of Arabic language understanding and comprehension, culture, and heritage. Note that some of these tasks are machine-translated. | Arabic (Some MT) |
```

haileyschoelkopf commented on lm_eval/tasks/README.md:
```suggestion
| [arabic_leaderboard_light](arabic_leaderboard_light/README.md) | A light version of the tasks in the Open Arabic LLM Leaderboard (i.e., 10% samples of the test set in the original benchmarks), focusing on the evaluation of models that reflect the characteristics of Arabic language understanding and comprehension, culture, and heritage. Note that some of these tasks are machine-translated. | Arabic (Some MT) |
```

haileyschoelkopf commented on lm_eval/tasks/arabic_leaderboard_complete/alghafa/alghafa.yaml:
```suggestion
  - arabic_leaderboard_alghafa_mcq_exams_test_ar
```
missing `_leaderboard` for these tasks still I believe?

haileyschoelkopf commented on lm_eval/tasks/arabic_leaderboard_complete/README.md:
Got it! Can we name your variant and its subtasks `arabic_leaderboard_arabic_mmlu...` then to make this clear?

haileyschoelkopf commented on lm_eval/tasks/arabic_leaderboard_complete/README.md:
perfect, lgtm!

haileyschoelkopf commented on lm_eval/tasks/arabic_leaderboard_complete/README.md:
Thanks! Think they should still also include `arabic_leaderboard` as a task name prefix

haileyschoelkopf commented on lm_eval/tasks/arabic_leaderboard_complete/README.md:
perfect! Let's also add `arabic_leaderboard_...` to prefix to these task names, unless you disagree?

haileyschoelkopf commented on lm_eval/tasks/arabic_leaderboard_complete/README.md:
got it!

haileyschoelkopf reviewed (changes requested):
Hi @shahrzads @Malikeh97 , I reviewed again! Just a few more requests re: standardizing task naming (sorry...) then we'd be good to go!

Also, would it be possible to run the precommit? (`pip install pre-commit`, `pre-commit install`)?

shahrzads commented on lm_eval/tasks/README.md:
Done

shahrzads commented on lm_eval/tasks/README.md:
Done

shahrzads commented on lm_eval/tasks/arabic_leaderboard_complete/alghafa/alghafa.yaml:
Done

shahrzads commented:
> Hi @shahrzads @Malikeh97 , I reviewed again! Just a few more requests re: standardizing task naming (sorry...) then we'd be good to go!
> 
> Also, would it be possible to run the precommit on all changed files? (`pip install pre-commit`, `pre-commit install`, then re-commit changes)?

Hi @haileyschoelkopf,

Thanks a lot for the review and your great comments! I have addressed all the comments, including standardizing the task naming.

I have also run the pre-commit on all the changed files, as requested.

Additionally, to make the benchmark more aligned with OALL (https://huggingface.co/spaces/OALL/Open-Arabic-LLM-Leaderboard) and lighteval (https://github.com/huggingface/lighteval), we added the Arabic mt_mmlu to the benchmark. The details are also added to the RADME file. Please take a look and let me know if there are any further changes needed.

Thanks again for your guidance!

shahrzads commented:
Hi @haileyschoelkopf,

Hope you are doing well!

I would like to follow up and ask for your feedback on the new changes. We would appreciate it if you can review the new version of the code and let us know your opinion.

shahrzads commented:
Hi @haileyschoelkopf,

Hope you are doing well!

I would like to follow up on addressing your feedback and ask for your comment on the new changes. We would appreciate it if you can review the new version of the code and let us know your opinion.

Malikeh97 commented:
Hi @haileyschoelkopf 
Thanks for all the constructive comments! Our PR is ready to merge. But, I see 2 workflows awaiting approval. Is there anything that we should done from our end?

Best,
Malikeh

lintangsutawika commented:
@Malikeh97 thanks for the PR!
